idl/idl-parser/src/org/jetbrains/dukat/idlParser/visitors/DefinitionVisitor.kt

```diff
@@ -0,0 +1,26 @@
+package org.jetbrains.dukat.idlParser.visitors
+
+import org.antlr.webidl.WebIDLBaseVisitor
+import org.antlr.webidl.WebIDLParser
+import org.jetbrains.dukat.idlDeclarations.IDLAttributeDeclaration
+import org.jetbrains.dukat.idlDeclarations.IDLDeclaration
+import org.jetbrains.dukat.idlDeclarations.IDLInterfaceDeclaration
+import org.jetbrains.dukat.idlParser.getName
+
+internal class DefinitionVisitor : WebIDLBaseVisitor<IDLDeclaration>() {
+    private val attributes: MutableList<IDLAttributeDeclaration> = mutableListOf()
+
+    override fun visitAttributeRest(ctx: WebIDLParser.AttributeRestContext): IDLDeclaration {
+        with (AttributeVisitor()) {
```

Comment:
`with` block return values so it (at least it looks like) it's better to have just
```typescript
attributes.add(with() {
    visit(ctx)
    visitAttributeRest(ctx)
})